### PR TITLE
feat: Add SentryCli js interface

### PR DESCRIPTION
**Pull request by HazAT**
Rename package to `@sentry/cli`
Add tests with jest
Add prettier
Add eslint

(Setup stolen from https://github.com/getsentry/zeus-cli thx @jan-auer)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,29 +1,16 @@
-var os = require('os');
-var childProcess = require('child_process');
-var pkgInfo = require('./package.json');
+const os = require('os');
+const pkgInfo = require('./package.json');
+const SentryCli = require('./js/cli');
 
-var path = null;
+exports.default = SentryCli;
```
Comment (jan-auer) [inline index.js]: Please use this instead:

```js
module.exports = SentryCli;
```

It is yet unusual in Node to use the ES6 style modules, especially in older versions.

File: index.js [hunk 2]
```diff
@@ -1,29 +1,16 @@
-var os = require('os');
-var childProcess = require('child_process');
-var pkgInfo = require('./package.json');
+const os = require('os');
+const pkgInfo = require('./package.json');
+const SentryCli = require('./js/cli');
```
Comment (jan-auer) [inline index.js]: `const` will break Node v4 and older unless specified with `'use strict';`.

File: package.json [hunk 3]
```diff
@@ -1,10 +1,17 @@
 {
-  "name": "sentry-cli-binary",
+  "name": "@sentry/cli",
   "version": "1.25.0",
   "description": "A command line utility to work with Sentry. https://docs.sentry.io/hosted/learn/cli/",
```
Comment (jan-auer) [inline package.json]: What is the minimum supported node version? We should add an `engines` field here.

File: .travis.yml [hunk 4]
```diff
@@ -18,6 +18,14 @@ matrix:
       osx_image: xcode6.4
       env: TARGET=x86_64-apple-darwin
 
+    # Test js
```
Comment (jan-auer) [inline .travis.yml]: Please add a test with the minimum supported Node version.

File: .eslintrc [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+    "extends": ["airbnb-base", "prettier"],
```
Comment (jan-auer) [inline .eslintrc]: Depending on the supported Node version, `eslint-config-airbnb-base` might suggest syntax that is not yet available. Maybe you should add a legacy config or es5-targeted config here instead and remove the overrides below?

Comment (jan-auer) [reply .eslintrc]: The config you declare here is compatible with Node v4, provided you put `'use strict';` at the top of every file.

File: package.json [hunk 6]
```diff
@@ -26,5 +33,26 @@
...
+    "eslint-config-airbnb-base": "^12.1.0",
+    "eslint-config-prettier": "^2.9.0",
+    "eslint-plugin-import": "^2.8.0",
+    "jest": "^21.2.1",
+    "npm-run-all": "^4.1.2",
+    "prettier": "^1.9.0",
+    "prettier-check": "^2.0.0"
+  },
+  "jest": {
+    "testRegex": "(/__tests__/.*|(\\.|/)(test|spec))\\.(jsx?|tsx?)$",
```
Comment (jan-auer) [inline package.json]: No need for all that, BUT you should use:

```json
{
  "collectCoverage": true,
  "testEnvironment": "node"
}
```

**Comment by jan-auer** [comment]
Tried to fix all the above issues. The only thing that remains is that Travis will fail once we push a new release. `install-sentry-cli` will try to fetch the bumped version from GitHub, but it is not there yet. Not sure how to test this properly...

**Comment by mitsuhiko** [comment]
🛳 it
